services/jobs/src/main/java/com/linkedin/openhouse/jobs/api/validator/impl/OpenHouseJobsApiValidator.java

```diff
@@ -40,25 +43,28 @@ public void validateCreateJob(CreateJobRequestBody createJobRequestBody) {
               "clusterId : provided %s, %s",
               createJobRequestBody.getClusterId(), ALPHA_NUM_UNDERSCORE_ERROR_MSG_HYPHEN_ALLOW));
     }
-    if (!createJobRequestBody.getJobConf().getExecutionConf().isEmpty()
-        && !JobConfValidator.executionConfigValidator(
-            createJobRequestBody.getJobConf().getExecutionConf())) {
-      validationFailures.add(
-          String.format(
-              "memory : provided %s, %s",
-              createJobRequestBody.getJobConf().getExecutionConf().get(JOB_MEMORY_CONFIG),
-              SPARK_MEMORY_FORMAT));
-    }
+    JobConfValidator.validate(createJobRequestBody.getJobConf(), validationFailures);
     if (!validationFailures.isEmpty()) {
       throw new RequestValidationFailureException(validationFailures);
     }
   }
 
   // inner class to validate jobConf fields.
   private static class JobConfValidator {
-    public static boolean executionConfigValidator(Map<String, String> executionConf) {
-      String memoryConfig = executionConf.getOrDefault(JOB_MEMORY_CONFIG, "");
-      return memoryConfig.matches(SPARK_MEMORY_FORMAT);
+    private static boolean validateExecutionConfig(Map<String, String> conf) {
+      String memory = conf.getOrDefault(JOB_MEMORY_CONFIG, "");
+      return memory.matches(SPARK_MEMORY_FORMAT);
+    }
+
+    private static void validate(@NonNull JobConf conf, List<String> validationFailures) {
+      final Map<String, String> executionConfig = conf.getExecutionConf();
+      if (MapUtils.isNotEmpty(executionConfig)
+          && !JobConfValidator.validateExecutionConfig(executionConfig)) {
+        validationFailures.add(
+            String.format(
+                "memory : provided %s, %s",
```

Comment:
Should be make the error message more readable something like "Invalid memory format [%s] provided. Correct format: [%s]". I see that similar format is followed in other places in this class.

Reply:
Fixed